src/mat.c

```diff
@@ -69,26 +69,29 @@
     (sizeof(MAT_MKTEMP_DIR) + sizeof(MAT_MKTEMP_TPL) + sizeof(MAT_MKTEMP_FILE) - 2)
 
 static char *
-Mat_mktemp(char *path_buf, char *dir_buf)
+Mat_mktemp(char *path_buf, size_t path_buf_size, char *dir_buf, size_t dir_buf_size)
```

Comment:
It might be good enought to assume `MAT_MKTEMP_BUF_SIZE` for both buffers to avoid passing buffer sizes.

Reply:
If you prefer, I can make that change.  But it's not like the extra param is going to be slower than the file system interaction...

Reply:
Yes, please go ahead. There is no strong reason to change the declaration of Mat_mktemp.

Be aware that I pushed to your branch.

Reply:
>Be aware that I pushed to your branch.

My git-foo is weak.  Presumably this means I should get your changes to my branch, right?  Will have to google that...

Reply:
Just git pull on your branch.

Reply:
OK done, hopefully I haven't clobbered whatever you changed...

Reply:
@tbeu did you have a chance to look this over again?